jetstream/stream_config.go

```diff
@@ -26,154 +26,356 @@ import (
-		DenyPurge            bool             `json:"deny_purge,omitempty"`
-		AllowRollup          bool             `json:"allow_rollup_hdrs,omitempty"`
-		Compression          StoreCompression `json:"compression"`
-		FirstSeq             uint64           `json:"first_seq,omitempty"`
-
-		// Allow applying a subject transform to incoming messages before doing anything else
+		// Name is the name of the stream. It is required and must be unique
+		// across the JetStream account.
+		Name string `json:"name"`
+
+		// Description is an optional description of the stream.
+		Description string `json:"description,omitempty"`
+
+		// Subjects is a list of subjects that the stream is listening on.
+		// Wildcards are supported. Subjects cannot be set if the stream is
+		// created as a mirror.
+		Subjects []string `json:"subjects,omitempty"`
+
+		// Retention defines the message retention policy for the stream.
+		// Defaults to LimitsPolicy.
+		Retention RetentionPolicy `json:"retention"`
+
+		// MaxConsumers specifies the maximum number of consumers allowed for
+		// the stream.
+		MaxConsumers int `json:"max_consumers"`
+
+		// MaxMsgs is the maximum number of messages the stream will store.
+		// After reaching the limit, stream adheres to the discard policy.
+		// Defaults to -1 (unlimited).
+		MaxMsgs int64 `json:"max_msgs"`
+
+		// MaxBytes is the maximum total size of messages the stream will store.
+		// After reaching the limit, stream adheres to the discard policy.
+		// Defaults to -1 (unlimited).
+		MaxBytes int64 `json:"max_bytes"`
+
+		// Discard defines the policy for handling messages when the stream
+		// reaches its limits in terms of number of messages or total bytes.
+		Discard DiscardPolicy `json:"discard"`
+
+		// DiscardNewPerSubject is a flag to enable discarding new messages per
+		// subject when limits are reached. Requires DiscardPolicy to be
+		// DiscardNew and the MaxMsgsPerSubject to be set.
+		DiscardNewPerSubject bool `json:"discard_new_per_subject,omitempty"`
+
+		// MaxAge is the maximum age of messages that the stream will retain.
+		MaxAge time.Duration `json:"max_age"`
+
+		// MaxMsgsPerSubject is the maximum number of messages per subject that
+		// the stream will retain.
+		MaxMsgsPerSubject int64 `json:"max_msgs_per_subject"`
+
+		// MaxMsgSize is the maximum size of any single message in the stream.
+		MaxMsgSize int32 `json:"max_msg_size,omitempty"`
+
+		// Storage specifies the type of storage backend used for the stream
+		// (file or memory).
+		Storage StorageType `json:"storage"`
+
+		// Replicas is the number of stream replicas in clustered JetStream.
+		// Defaults to 1, maximum is 5.
+		Replicas int `json:"num_replicas"`
+
+		// NoAck is a flag to disable acknowledging messages received by this
+		// stream.
+		NoAck bool `json:"no_ack,omitempty"`
+
+		// Duplicates is the window within which to track duplicate messages.
+		// Defaults to 0.
+		Duplicates time.Duration `json:"duplicate_window,omitempty"`
```

Comment:
defaults to 2 minutes on the server though? not sure if you are telegraphing those here for users to know

Reply:
Yes I am - thanks, this was a mistake.

Reply:
Fixed